### ISSUE TITLE
vxlan - Containers with host networking is not able to ping containers across host.

### ISSUE THREAD

**Issue by sangeethah** [rancher/rancher#6755]
Rancher Version: v1.2.0-pre4-rc14

Networking - vxlan

Steps to reproduce the problem:

Create a stack.
Create a service in host network mode.
Create an external service pointing to managed ip address of 2 containers in different hosts.

From with in host network container , try to ping external service by name. It resolves to the ipaddress of the external service. But the ip address is not reachable. 

```
root@sangeerc14-10acre-1:/# dig test232160.test295993.rancher.internal +short
10.42.62.213
10.42.116.197
root@sangeerc14-10acre-1:/# ping 10.42.62.213
PING 10.42.62.213 (10.42.62.213) 56(84) bytes of data.
^C
--- 10.42.62.213 ping statistics ---
3 packets transmitted, 0 received, 100% packet loss, time 2017ms

root@sangeerc14-10acre-1:/#
```

```
=================================== FAILURES ===================================
_____________ test_dns_discoverys_with_hostnetwork_externalService _____________

admin_client = <cattle.Client object at 0x110a60250>
client = <cattle.Client object at 0x112098110>

    def test_dns_discoverys_with_hostnetwork_externalService(admin_client, client):
    
        # Verify if able to resolve external services from containers
        # that belong to another service in host network.
    
        port = "422"
        env, service, ext_service, con_list = \
            create_env_with_ext_svc(client, 1, port)
    
        launch_config_svc = {"imageUuid": SSH_IMAGE_UUID_HOSTNET,
                             "networkMode": "host",
                             "labels": dns_labels}
        random_name = random_str()
        service_name = random_name.replace("-", "")
        host_service = client.create_service(name=service_name,
                                             stackId=env.id,
                                             launchConfig=launch_config_svc,
                                             scale=1)
        host_service = client.wait_success(host_service)
        host_service.activate()
        ext_service.activate()
        host_service = client.wait_success(host_service, SERVICE_WAIT_TIMEOUT)
        ext_service = client.wait_success(ext_service, SERVICE_WAIT_TIMEOUT)
        assert host_service.state == "active"
        assert ext_service.state == "active"
    
        validate_external_service(
            admin_client, host_service, [ext_service], 33, con_list,
>           fqdn="." + env.name + ".rancher.internal")

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_service_dns_discovery.py:490: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

admin_client = <cattle.Client object at 0x110a60250>
service = {'createdTS': 1479954072000, 'launchConfig': {'tty': False, 'networkMode': u'host', 'vcpu': 1, 'labels': {'io.rancher....1-24T02:21:12Z', 'scalePolicy': None, 'fqdn': None, 'createIndex': 1, 'instanceIds': [u'1i1939'], 'selectorLink': None}
ext_services = [{'createdTS': 1479954072000, 'launchConfig': {}, 'accountId': u'1a5', 'secondaryLaunchConfigs': [], 'upgrade': None, ...nd': u'externalService', 'name': u'test232160', 'created': u'2016-11-24T02:21:12Z', 'fqdn': None, 'instanceIds': None}]
exposed_port = 33
container_list = [{'tty': False, 'registryCredentialId': None, 'labels': {'io.rancher.container.uuid': u'b334c853-5441-495e-96af-279417...mmand': None, 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': [u'rancher.internal']}]
exclude_instance = None, exclude_instance_purged = False
fqdn = '.test295993.rancher.internal'

    def validate_external_service(admin_client, service, ext_services,
                                  exposed_port, container_list,
                                  exclude_instance=None,
                                  exclude_instance_purged=False,
                                  fqdn=None):
        time.sleep(sleep_interval)
    
        containers = get_service_container_list(admin_client, service)
        assert len(containers) == service.scale
        for container in containers:
            print "Validation for container -" + str(container.name)
            host = admin_client.by_id('host', container.hosts[0].id)
            for ext_service in ext_services:
                expected_dns_list = []
                expected_link_response = []
                dns_response = []
                for con in container_list:
                    if (exclude_instance is not None) \
                            and (con.id == exclude_instance.id):
                        print "Excluded from DNS and wget list:" + con.name
                    else:
                        expected_dns_list.append(con.primaryIpAddress)
                        expected_link_response.append(con.externalId[:12])
    
                print "Expected dig response List" + str(expected_dns_list)
                print "Expected wget response List" + str(expected_link_response)
    
                # Validate port mapping
                ssh = paramiko.SSHClient()
                ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
                ssh.connect(host.ipAddresses()[0].address, username="root",
                            password="root", port=int(exposed_port))
    
                ext_service_name = ext_service.name
                if fqdn is not None:
                    ext_service_name += fqdn
                # Validate link containers
                cmd = "wget -O result.txt --timeout=20 --tries=1 http://" + \
                      ext_service_name + ":80/name.html;cat result.txt"
                print cmd
                stdin, stdout, stderr = ssh.exec_command(cmd)
    
                response = stdout.readlines()
>               assert len(response) == 1
E               assert 0 == 1
E                +  where 0 = len([])

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1336: AssertionError
----------------------------- Captured stdout call -----------------------------
Validation for container -test295993-test520777-1
Expected dig response List[u'10.42.116.197', u'10.42.62.213']
Expected wget response List[u'8e4980a2ef0d', u'd0589e2de8b1']
wget -O result.txt --timeout=20 --tries=1 http://test232160.test295993.rancher.internal:80/name.html;cat result.txt
 24 tests deselected by '-k test_dns_discoverys_with_hostnetwork_externalService' 
============== 1 failed, 1 passed, 24 deselected in 71.47 seconds ==============

Process finished with exit code 0

admin_client = <cattle.Client object at 0x110a60250>
client = <cattle.Client object at 0x112098110>

    def test_dns_discoverys_with_hostnetwork_externalService(admin_client, client):
    
        # Verify if able to resolve external services from containers
        # that belong to another service in host network.
    
        port = "422"
        env, service, ext_service, con_list = \
            create_env_with_ext_svc(client, 1, port)
    
        launch_config_svc = {"imageUuid": SSH_IMAGE_UUID_HOSTNET,
                             "networkMode": "host",
                             "labels": dns_labels}
        random_name = random_str()
        service_name = random_name.replace("-", "")
        host_service = client.create_service(name=service_name,
                                             stackId=env.id,
                                             launchConfig=launch_config_svc,
                                             scale=1)
        host_service = client.wait_success(host_service)
        host_service.activate()
        ext_service.activate()
        host_service = client.wait_success(host_service, SERVICE_WAIT_TIMEOUT)
        ext_service = client.wait_success(ext_service, SERVICE_WAIT_TIMEOUT)
        assert host_service.state == "active"
        assert ext_service.state == "active"
    
        validate_external_service(
            admin_client, host_service, [ext_service], 33, con_list,
>           fqdn="." + env.name + ".rancher.internal")

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/test_service_dns_discovery.py:490: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

admin_client = <cattle.Client object at 0x110a60250>
service = {'createdTS': 1479954072000, 'launchConfig': {'tty': False, 'networkMode': u'host', 'vcpu': 1, 'labels': {'io.rancher....1-24T02:21:12Z', 'scalePolicy': None, 'fqdn': None, 'createIndex': 1, 'instanceIds': [u'1i1939'], 'selectorLink': None}
ext_services = [{'createdTS': 1479954072000, 'launchConfig': {}, 'accountId': u'1a5', 'secondaryLaunchConfigs': [], 'upgrade': None, ...nd': u'externalService', 'name': u'test232160', 'created': u'2016-11-24T02:21:12Z', 'fqdn': None, 'instanceIds': None}]
exposed_port = 33
container_list = [{'tty': False, 'registryCredentialId': None, 'labels': {'io.rancher.container.uuid': u'b334c853-5441-495e-96af-279417...mmand': None, 'serviceIds': None, 'deploymentUnitUuid': None, 'createIndex': None, 'dnsSearch': [u'rancher.internal']}]
exclude_instance = None, exclude_instance_purged = False
fqdn = '.test295993.rancher.internal'

    def validate_external_service(admin_client, service, ext_services,
                                  exposed_port, container_list,
                                  exclude_instance=None,
                                  exclude_instance_purged=False,
                                  fqdn=None):
        time.sleep(sleep_interval)
    
        containers = get_service_container_list(admin_client, service)
        assert len(containers) == service.scale
        for container in containers:
            print "Validation for container -" + str(container.name)
            host = admin_client.by_id('host', container.hosts[0].id)
            for ext_service in ext_services:
                expected_dns_list = []
                expected_link_response = []
                dns_response = []
                for con in container_list:
                    if (exclude_instance is not None) \
                            and (con.id == exclude_instance.id):
                        print "Excluded from DNS and wget list:" + con.name
                    else:
                        expected_dns_list.append(con.primaryIpAddress)
                        expected_link_response.append(con.externalId[:12])
    
                print "Expected dig response List" + str(expected_dns_list)
                print "Expected wget response List" + str(expected_link_response)
    
                # Validate port mapping
                ssh = paramiko.SSHClient()
                ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
                ssh.connect(host.ipAddresses()[0].address, username="root",
                            password="root", port=int(exposed_port))
    
                ext_service_name = ext_service.name
                if fqdn is not None:
                    ext_service_name += fqdn
                # Validate link containers
                cmd = "wget -O result.txt --timeout=20 --tries=1 http://" + \
                      ext_service_name + ":80/name.html;cat result.txt"
                print cmd
                stdin, stdout, stderr = ssh.exec_command(cmd)
    
                response = stdout.readlines()
>               assert len(response) == 1
E               assert 0 == 1
E                +  where 0 = len([])

/Users/sangeethahariharan1/cattle/validation/validation-tests/tests/v2_validation/cattlevalidationtest/core/common_fixtures.py:1336: AssertionError
```

**Comment by leodotcloud** [comment]
Fixed in version: `rancher/net:v0.9.3`

**Comment by sangeethah** [comment]
Rancher Version: Build from master

Networking - vxlan with rancher/net:v0.9.3

Create a stack.
Create a service in host network mode.
Create an external service pointing to managed ip address of 3 containers in 3 different hosts in the setup.

From with in host network container , try to ping external service by name. It resolves to 3 ipaddress of the external service. 

Able to reach all the containers (in managed network) from this container in host network.
